ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/utils/useExternalPlugins.ts

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from 'react';
+import { PluginStates } from 'lib/services/plugins/reducer';
+import { contentScriptsToCodeMirrorPlugin } from 'lib/services/plugins/utils/loadContentScripts';
+import { resolve, extname } from 'path';
+
+
+export default function useExternalPlugins(CodeMirror: any, plugins: PluginStates) {
+
+	const [options, setOptions] = useState({});
+	useEffect(() => {
+		let newOptions = {};
+
+		const contentScripts = contentScriptsToCodeMirrorPlugin(plugins);
+
+		for (const contentScript of contentScripts) {
+			const mod = contentScript.module;
+			for (const addon of mod.addons) {
+				require(`codemirror/addon/${addon}`);
```

Comment:
If this is an external plugin, shouldn't it be at the plugin level that we require whatever is needed. What if someone create a brand new plugin that's not in codemirror/addon, I guess it wouldn't work in this case?

Reply:
In fact, it might help to add the documentation to the `ContentScriptType` interface, like it's done for the markdown-it. As knowing how it should work will help discuss the implementation details.

Reply:
These are specifically for the [codemirror addons](https://codemirror.net/doc/manual.html#addons). Codemirror has an odd way of loading plugins that means we can't load them through the contrentScript plugin. Basically when you require a CodeMirror plugin it modifies the state of the already imported codemirror instance. All other requires can be done at the plugin level, so I think this separation makes sense.

I'll get on that documentation soon and hopefully that will help things makes sense.

Reply:
Ok that makes sense. Is there also a concept of external addon in CodeMirror (one that wouldn't already be packaged with the distribution)?

Reply:
They exist, but I don't know of many. From what I have seen they usually don't use the special loading system that the codemirror built-in addons use. The system that we use internally of calling a function with `CodeMirror` as an argument seems to be normal. Our content scripts work exactly like this and pass `CodeMirror` to the function. This would allow plugin developers to require their own external plugin and pass `CodeMirror` on to it.